src/views/UpdatePassword.js

```diff
@@ -0,0 +1,132 @@
+import React, { Component } from "react";
+import PropTypes from "prop-types";
+import { connect } from "react-redux";
+import TextInput from "../components/common/Inputs/TextInput";
+import FormButton from "../components/common/Buttons/FormButton";
+import {
+  handleInputChange,
+  handleUpdatePassword
+} from "../redux/actions/updatePasswordActions";
+import Validator from "../utils/validator";
+import { isEmpty } from "../utils/helperFunctions";
+
+export class UpdatePassword extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      errors: {}
+    };
+    this.handleOnChange = this.handleOnChange.bind(this);
+    this.handleSubmit = this.handleSubmit.bind(this);
+  }
+
+  componentWillReceiveProps(nextProps) {
+    const { errors } = nextProps;
+    this.setState({
+      errors
+    });
+  }
+
+  handleOnChange(e) {
+    const { value, name } = e.target;
+    const { handleInputChange: onChangeAction } = this.props;
+    onChangeAction(name, value);
+  }
+
+  handleSubmit() {
+    const {
+      password,
+      confirmPassword,
+      handleUpdatePassword: submitNewPassword
+    } = this.props;
+    const errors = {
+      ...Validator.formData({ password, confirmPassword }),
+      ...Validator.isMatch("password", password, confirmPassword)
+    };
+    if (!isEmpty(errors)) {
+      this.setState({
```

Comment:
Why are we setting the state in the component?
Is it possible to let redux manage all the state of the component?